### PR TITLE
Crd scaling test

### PR DESCRIPTION
**Pull request by aabughosh**

### CODE REVIEW

File: scripts/deploy-operator-crd-scaling.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
```
Comment (edcdavid) [inline scripts/deploy-operator-crd-scaling.sh:1]: this file needs executable rights

Comment (rdavid) [reply scripts/deploy-operator-crd-scaling.sh:1]: And the new line at the end :-)

File: Makefile [hunk 2]
```diff
@@ -111,3 +111,6 @@ deploy-services:
 # delete services
 delete-services:
 	./scripts/manage-service.sh delete
+
+deploy-test-crd-scaling-operator:
+	./scripts/deploy-operator-crd-scaling.sh
```
Comment (edcdavid) [inline Makefile:116]: also needs to be added in the install target

**Review by edcdavid — COMMENTED** [review]
Scaling does not seem to work (have not debugged it yet, rights issue?)

File: Makefile [hunk 3]
```diff
@@ -111,3 +112,6 @@ deploy-services:
...
```
Comment (rdavid) [inline Makefile:117]: Please add newline to the end of the file.

File: scripts/deploy-operator-crd-scaling.sh [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+gcc -g -Wall bin/controller-gen.c -o bin/bin/controller-gen
+
+make manifests
+make install
+make deploy IMG=quay.io/testnetworkfunction/crd-operator-scaling:latest
+
+oc wait deployment new-pro-controller-manager -n "$TNF_EXAMPLE_CNF_NAMESPACE" --for=condition=available --timeout=240s
+
+make addrole
+kubectl apply -f config/samples  --validate=false
```
Comment (rdavid) [inline scripts/deploy-operator-crd-scaling.sh:26]: No need for double spaces.

**Review by rdavid — APPROVED** [review]

**Review by edcdavid — COMMENTED** [review]
add a delay in the controller when scaling the replicas up or down, so that a tester has time to check it on the cli. for instance 10 seconds

**Review by edcdavid — COMMENTED** [review]
Add a label tnf test label in the template section of the 'jack' deployment so that the test can pickup the deployment
  "template": {
    "metadata": {
      "creationTimestamp": null,
      "labels": {
        "app": "jack",
        "test-network-function.com/generic": "target"
      }
    },

File: scripts/deploy-operator-crd-scaling.sh [hunk 5]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+
+
+SCRIPT_DIR=$(dirname "$0")
+source "$SCRIPT_DIR"/init-env.sh
+CRD_SCALING_URL="https://github.com/test-network-function/crd-operator-scaling.git"
+git clone $CRD_SCALING_URL
```
Comment (edcdavid) [inline scripts/deploy-operator-crd-scaling.sh:7]: delete any previous "crd-operator-scaling" directory before cloning

**Review by edcdavid — APPROVED** [review]

**Pull request merged**